File: frontend/react-app/src/api/dto/ChatRoomDto.js
import MessageDto from "./MessageDto";

export default class ChatRoomDto {
    constructor(userId, userName, userSurname, lastMessage, chatId) {
        this.userId = userId;
        this.userName = userName;
        this.userSurname = userSurname;
        this.lastMessage = lastMessage;
        this.chatId = chatId;
    }

    static fromJSON(data) {
        return new ChatRoomDto(
            data.userId,
            data.userName,
            data.userSurname ? data.userSurname : "",
            data.lastMessage && MessageDto.fromJSON(data.lastMessage),
            data.chatId
        );
    }
}

File: backend/Service/src/main/java/ru/nikidzawa/db/store/dto/object/dto/ChatRoomDto.java
package ru.nikidzawa.db.store.dto.object.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatRoomDto {
    Long userId;
    String userName;
    String userSurname;

    MessageDto lastMessage;

    Long chatId;
    Boolean isGroupChat;
    String chatName;
}
